File: server/DTO/FriendDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace server.DTO
{
    public class FriendDTO
    {
        public string username { get; set; }
        public string avatar { get; set; }
        public int friendId { get; set; }
        public int userId { get; set; }
    }
}

File: client/lib/data/dto/friend_dto.dart
class FriendDTO {
  final String? username;
  final String? avatar;
  final int? friendId;
  final int? userId;

  FriendDTO({
    this.username,
    this.avatar,
    this.friendId,
    this.userId,
  });

  factory FriendDTO.fromJson(Map<String, dynamic> json) {
    return FriendDTO(
      username: json['username'] as String?,
      avatar: json['avatar'] as String?,
      friendId: json['friendId'] as int?,
      userId: json['userId'] as int?,
    );
  }
}